rmgpy/rmg/model.py

```diff
@@ -144,12 +144,22 @@ def processThermoData(self, database, thermo0, thermoClass=NASA):
 
         # Add on solvation correction
         if Species.solventData and not "Liquid thermo library" in thermo0.comment:
-            #logging.info("Making solvent correction for {0}".format(Species.solventName))
             soluteData = database.solvation.getSoluteData(self)
-            solvation_correction298 = database.solvation.getSolvationCorrection298(soluteData, Species.solventData)
-            # correction is added to the entropy and enthalpy
-            wilhoit.S0.value_si = (wilhoit.S0.value_si + solvation_correction298.entropy)
-            wilhoit.H0.value_si = (wilhoit.H0.value_si + solvation_correction298.enthalpy)
+            if self.isSolventinCoolProp:
+                # If the solvent data are available in CoolProp, the more accurate temperature dependence of the solvation free
+                # energy is applied to make the thermo correction
+                T = Species.rxnTemp.value_si # reaction temperature in K
+                solvation_correction = database.solvation.getSolvationCorrection(self, soluteData, T)
+                # correction is added to the entropy and enthalpy
+                wilhoit.S0.value_si = (wilhoit.S0.value_si + solvation_correction.entropy)
```

Comment:
both the if and the else have the same lines `wilhoit.S0.value_si = (wilhoit.S0.value_si +...`: refactor them to only appear once, _after_ the conditional.